python/meterpreter/ext_server_stdapi.py

```diff
@@ -1383,11 +1384,17 @@ def stdapi_sys_process_execute(request, response):
         proc_h.start()
     else:
         proc_h = subprocess.Popen(args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    proc_h_id = meterpreter.add_process(proc_h)
+
+    win32_handle = None
+    if has_windll:
+        k32 = ctypes.windll.kernel32
+        win32_handle = k32.OpenProcess(PROCESS_ALL_ACCESS, False, proc_h.pid)
```

Comment:
The return value from[ `OpenProcess`](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-openprocess) is a [`HANDLE`](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/929187f0-f25c-4b05-9497-16b066d8a912) which is `void*`. OpenProcess should have it's args and restype defined for that. If the handle needs the high 32-bits on a 64-bit system, this would have an issue.

Reply:
Taking my suggestions below from `add_process` you'd change this back to `proc_h_id = meterpreter.add_process(proc_h)`.  Now `proc_h_id` is the windows HANDLE and is already the value used by `TLV_TYPE_PROCESS_HANDLE` below.

When you go to add the channel, you'd do:
```
channel_id = meterpreter.add_channel(MeterpreterProcess(proc_h, proc_h_id))
```